src/hooks/useGasPrice.ts

```diff
@@ -45,7 +74,11 @@ const fetchGasOracle = async (gasPriceOracle: GasPriceOracle): Promise<bigint> =
 
   const json = await response.json()
   const data = json.data || json.result || json
-  return BigInt(data[gasParameter] * Number(gweiFactor))
+
+  if (uri.startsWith(ETHER_SCAN_URL)) {
```

Comment:
I would do duck typing instead of checking the URL. This will work even of the URL change, and it will also protect you if the API changes.

```suggestion
  if (isEtherscanResult(data)) {
    return parseEtherscanOracleResult(result)
  }
```

```js
function isEtherscanResult(data): data is EtherscanResult {
  return 'FastGasPrice' in data && 'suggestBaseFee' in data
}
```